lib/sisimai/bite/email/imailserver.rb

```diff
@@ -0,0 +1,181 @@
+    # Sisimai::Bite::Email::IMailServer parses a bounce email which created by
+    # Ipswitch IMail Server.
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite::Email/IMailServer.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :'x-mailer' => %r/\A[<]SMTP32 v[\d.]+[>][ ]*\z/,
+        :'subject'  => %r/\AUndeliverable Mail[ ]*\z/,
+      }
+      Re1 = {
+        :begin  => %r/\A\z/,    # Blank line
+        :error  => %r/Body of message generated response:/,
+        :rfc822 => %r/\AOriginal message follows[.]\z/,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReSMTP = {
+        conn: %r{(?:
+             SMTP[ ]connection[ ]failed,
+            |Unexpected[ ]connection[ ]response[ ]from[ ]server:
+            )
+        },
+        ehlo: %r|Unexpected response to EHLO/HELO:|,
+        mail: %r|Server response to MAIL FROM:|,
+        rcpt: %r|Additional RCPT TO generated following response:|,
+        data: %r|DATA command generated response:|,
+      }
+      ReFailure = {
+        hostunknown: %r/Unknown[ ]host/x,
+        userunknown: %r/\A(?:Unknown[ ]user|Invalid[ ]final[ ]delivery[ ]userid)/x,
+        mailboxfull: %r/\AUser[ ]mailbox[ ]exceeds[ ]allowed[ ]size/x,
+        securityerr: %r/\ARequested[ ]action[ ]not[ ]taken:[ ]virus[ ]detected/x,
+        undefined:   %r/\Aundeliverable[ ]to[ ]/x,
+        expired:     %r/\ADelivery[ ]failed[ ]\d+[ ]attempts/x,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'IPSWITCH IMail Server'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['X-Mailer']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from IMailServer
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['subject'] =~ Re0[:subject]
+        match += 1 if mhead['x-mailer'] && mhead['x-mailer'] =~ Re0[:'x-mailer']
+        return nil if match.zero?
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [97.47/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->